src/main/java/Duke.java

```diff
@@ -112,12 +110,12 @@ public void run() {
                             } catch (DukeException exp) {
                                 ui.showErrorMsg(exp.getMessage());
                             }
-                        } else if (c.getCommandType().equals("todo")) {
+                        } else if (cmd.getCommandType().equals("todo")) {
                             try {
-                                if (c.getInstruction().equals("") || c.getInstruction().equals("todo")) {
+                                if (cmd.getInstruction().equals("") || cmd.getInstruction().equals("todo")) {
                                     throw new DukeException("☹ OOPS!!! The description of todo cannot be empty");
                                 } else {
-                                    tasks.addItemToList(new Todo(c.getInstruction()));
+                                    tasks.addItemToList(new Todo(cmd.getInstruction()));
                                     try {
```

Comment:
Perhaps you could reduce the lines and improve clarity by removing the multiple try catch clauses into one larger try bracket in the method? Also, another approach is to maybe have all methods throw exceptions and have one class handle all exceptions?